### ISSUE TITLE
OpenLDAP user filter and multi-valued fields

### ISSUE THREAD

**Issue by gyp5y** [privacyidea/privacyidea#556]
PrivacyIdea 2.15 Debian apt.
LDAP resolver not using User filter configured. Filters are hardcoded and there's no way to match users by mail or uid e.g. (|(mail=%s)(uid=%s))
Second issue with resolver is that it's not passing multi-valued fields in results. There's hardcoded 'mobile' field and no way to add more fields like 'memberOf' or 'objectClass'

**Comment by cornelinux** [comment]
It is not a usual setup to give multiple usernames to a user object. This is why you can only have *one* %s tag in the filter. You can however add a second resolver and use the forward token to assign the same token to the 2nd user object.

Using the mapping you can assign new attributes.

**Comment by gyp5y** [comment]
It's not unusual. It's a way to make possible for user to authenticate with his email address(-es if match in proxyAdresses) and/or username.
How about second issue I've mentioned?

**Comment by cornelinux** [comment]
Your second issue is it about arbitrary attributes (which works fine) or about attributes being a list.

What is your use case?

**Comment by cornelinux** [comment]
Well, you can use multiple usernames with multiple revolvers. 
At the moment I am not sure, if this is only about extending the filter.  Would have to check this.

In future: please open one issue per topic. Thanks
.

**Comment by gyp5y** [comment]
Second issue is about fields like 'memberOf' which has multiple values in response from LDAP. In LDAP resolver there's hardcoded 'mobile' attribute and it's working just like it should and returning a list of values. I've added additional attributes in my setup but it would be great if it will be moved to config.
Regarding multiple usernames - it's just about modifying filter. There's "User Filter" option in resolver config (I think it should do just what I'm trying to) but it's not used in code.
I've changed filter to "(&{0!s}(|(proxyAddresses={2!s})(|(mail={2!s})({1!s}={2!s}))))" and it's now working fine but it's not suitable for production. I think this option is needed 'cos most user storages (e.g. SimplesamlPhp) has an option to choose what attributes to use searching for a user by login name.
